Cargo.toml

```diff
@@ -61,7 +61,7 @@ gix = { version = "0.67.0", default-features = false, features = [
     "attributes",
     "blob-diff",
     "index",
-    "max-performance-safe",
+    "max-performance",
```

Comment:
I'm slightly worried about whether this will compile correctly as a static linux-musl binary, like our release build. Did you check?

Perhaps we should have a presubmit check that checks that directly. I'm not sure the `--all-features` check is sufficient (though the build that would happen after the PR is merged would probably check it). Or do we have one that I forgot about?

Reply:
Oh, I didn't know that release builds aren't triggered by PRs.

The x86_64 musl binary works fine.

Reply:
I think that's a general concern about including C dependency.

FWIW, the latest libz-ng-sys has a linking issue with static msvcrt, which was detected by CI.

Reply:
Yeah, I see your note about downgrading it. I wasn't really too worried about musl in particular since I think `zlib-ng` is used in that part of the world, but the MSVC one is more interesting.

Well, I'll see if I can try to fix that static linking issue, but the changes between the current version and the working one will need a bit of review and investigation (lots of "commit soup" it seems...)